[Pygen]: Add README.md for PyFlow

aneels3 opened the pull request:
Signed-off-by: aneels3 <b150023ec@nitsikkim.ac.in>

hodjat91 commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,74 @@
+## Overview
+
+RISCV-DV-PyFlow is a python based open-source instruction generator for RISC-V
```
a purely python based

hodjat91 commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,74 @@
+## Overview
+
+RISCV-DV-PyFlow is a python based open-source instruction generator for RISC-V
+processor verification. It currently supports the following features:
```
We should also talk about / refer to pyvsc as the main tool/library that we're using.

aneels3 replied:
Good point.
I have added it with the pyvsc repo link.

hodjat91 commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,74 @@
+processor verification. It currently supports the following features:
+
+- Supported instruction set: RV32IMAFDC
+- Supported privileged mode: machine mode
```
Supported privileged modes: for now only machine mode is supported.

hodjat91 commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,74 @@
+- Supports mixing directed instructions with random instruction stream
+- Support for direct & vectored interrupt table.
+- Multi-hart support
+- Instruction generation coverage model(Currently supports RV32I and RV32M only)
```
Functional coverage framework (reports GUI as well as text, currently supports RV32I and RV32M only)

hodjat91 commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,74 @@
+- Support for direct & vectored interrupt table.
+- Multi-hart support
+- Instruction generation coverage model(Currently supports RV32I and RV32M only)
+- Co-simulation with multiple ISS : spike, riscv-ovpsim
```
Supported ISS: Spike, Ovpsim

hodjat91 commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,74 @@
+- Supported instruction set: RV32IMAFDC
+- Supported privileged mode: machine mode
+- Illegal instruction and HINT instruction generation
+- Random forward/backward branch instructions
```
loops are also supported, right? Mind adding it here?

hodjat91 commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,74 @@
+### Prerequisites
+
+To be able to run the generator, you need to have RISCV-GCC compiler toolchain and ISS
+(Instruction Set Simulator) installed.
```
(Instruction Set Simulator) installed (Spike is preferred)

hodjat91 commented on pygen/pygen_src/README.md:
````diff
@@ -0,0 +1,74 @@
+```
+
+## Note
+Time to generate more than 10k instructions for single iteration is around 10-12 minutes.
````
Currently, time to generate a single program with larger than 10k instructions is around x minutes (let's be more precise, maybe 10-11?). We are working on improving the overall performance.

hodjat91 reviewed (commented):
Hi @aneels3 

Thanks for the CL. PTAL at the comments and let me know if there is any questions.

aneels3 commented:
> Hi @aneels3
> 
> Thanks for the CL. PTAL at the comments and let me know if there is any questions.

Hi @hodjat91 
I have addressed all the comments.
Please check it out.
Let me know if it needs more modification.

udinator commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,76 @@
+## Overview
+
+RISCV-DV-PyFlow is a purely python based open-source instruction generator for RISC-V processor
```
nit: capitalize `python` here and throughout

hodjat91 reviewed (approved):
LGTM, please address Udi's comment.

udinator commented on pygen/pygen_src/README.md:
```diff
@@ -0,0 +1,76 @@
+## Overview
+
+RISCV-DV-PyFlow is a purely PYTHON based open-source instruction generator for RISC-V processor
```
nit: sorry for the confusion, I meant `python` -> `Python`

aneels3 replied:
oops!
Fixed now

udinator reviewed (approved):
LGTM, thanks @aneels3 !
